lib/inspec/resources/sys_info.rb

```diff
@@ -3,117 +3,140 @@
 
 module Inspec::Resources
   # this resource returns additional system informatio
-  class System < Inspec.resource(1)
+  class SysInfo < Inspec.resource(1)
     name "sys_info"
-    supports platform: "unix"
+    supports platform: "linux"
     supports platform: "windows"
 
-    desc "Use the user InSpec system resource to test for operating system properties."
+    desc "Use the sys_info resource to test for operating system properties."
     example <<~EXAMPLE
       describe sys_info do
-        its('hostname') { should eq 'example.com' }
+        its('hostname') { should eq 'hostname' }
       end
 
       describe sys_info do
-        its('fqdn') { should eq 'user.example.com' }
+        its('fqdn') { should eq 'hostname.example.com' }
       end
 
     EXAMPLE
 
-    %w{ domain fqdn ip_address short }.each do |opt|
-      define_method(opt.to_sym) do
-        hostname(opt)
-      end
-    end
-
-    # returns the hostname of the local system
-    def hostname(opt = nil)
+    def initialize
       os = inspec.os
+      @cache = nil
+
       if os.linux?
-        linux_hostname(opt)
-      elsif os.darwin?
-        mac_hostname(opt)
+        @cache = LinuxInfo.new(inspec)
```

Comment:
While you have a ivar named `@cache`, it actually isn't being used as a cache - you're repeatedly calling `info()` on it, which is expensive. The return value of info is what you want to cache.

Consider modifying this to:
```
@cache = LinuxInfo.new(inspec).info
```

and then (for example)

```
    def domain
      @cache[:domain]
    end
```